### PR TITLE
Software Engineering ladder refinements (based on feedback from EMs)

### PR DESCRIPTION
**Pull request by Waldo000000**
This PR proposes refinements to the Software Engineering ladder, based on suggestions from Engineering Managers following the most recent performance review cycle.

These refinements should not affect the way that anyone is evaluated in their next performance review; they should only result in _existing expectations_ being written down and understood more clearly and consistently across all of Engineering.

We plan to follow a similar process after each performance cycle going forward. To understand more about the principles and process behind  these refinements, reach out to your manager, or see:
- [How we use calibration to refine our role ladders](https://octopushq.atlassian.net/wiki/spaces/RND/pages/2236940315/Engineering+calibration+process#How-we-use-calibration-to-refine-our-role-ladders) (internal wiki link)

If you are on this ladder and have any concerns or feedback about the proposed refinements, please reach out to your manager.

- [x] 2022-07-20 Propose post-calibration role ladder refinement process ([internal link](https://octopushq.atlassian.net/wiki/spaces/RND/pages/2236940315/Engineering+calibration+process#How-we-use-calibration-to-refine-our-role-ladders))
- [x] 2022-07-27 Hold post-calibration role ladder refinement workshop (collate suggestions and voting from Engineering Managers)
- [x] 2022-08-04 Incorporate highest-voted suggestions in this PR
- [x] 2022-08-18 Collate any further feedback via EMs, then merge PR

### CODE REVIEW

File: Software-Engineering/L2-Software-Engineer.md [hunk 1]
```diff
@@ -1,6 +1,6 @@
 # L2: Software Engineer
 
-> _I am a valuable and enthusiastic member of an Octopus team. I can work independently on small projects, and am focused on learning with a trajectory of growth into the contribution of a Senior Software Engineer or beyond._
+> _I am a valuable and enthusiastic member of an Octopus team. I can work independently on small projects, and am focused on learning with a trajectory of growth into the contribution of a Senior Software Engineer or beyond. I still require some support from L3s to solve more difficult technical problems that are beyond my current experience level._
```
Comment (mattocto) [inline Software-Engineering/L2-Software-Engineer.md:3]: I found the use of `L3s` jarring here. Would using the term `Senior Engineer` be a better fit?

**Review by mattocto — COMMENTED** [review]
Just one minor comment. Overall this looks like a really positive change, helping to be explicit about some of the built-in assumptions in the career framework, and also providing some nice new examples of what this looks like in action.

File: Software-Engineering/L2-Software-Engineer.md (hunk 1 above)
Comment (cstrzadala) [reply Software-Engineering/L2-Software-Engineer.md:3]: +1 on this. I think it would read easier as well with `senior engineers` instead of `L3s`

**Review by MarkSiedle — APPROVED** [review]
Thanks Roy. This looks consistent with the outcomes/information captured in the meeting 🙇 

LGTM

**Review by cstrzadala — COMMENTED** [review]
One comment as per @mattocto . Outside that, LGTM.

**Review by mattocto — APPROVED** [review]
Looks good :)

**Review by kengel100 — APPROVED** [review]

**Pull request merged**